src/app/layout.tsx

```diff
@@ -65,6 +68,27 @@ type Props = { children: ReactNode };
 const RootLayout: FC<Props> = ({ children }) => {
   return (
     <html lang="en" suppressHydrationWarning>
+      <head>
+        {isProd && (
+          <>
+            {/* Google Analytics Measurement ID*/}
+            <script async src={gtag} />
+            {/* Inject the GA tracking code with the Measurement ID */}
+            <script
+              dangerouslySetInnerHTML={{
+                __html: `
+                window.dataLayer = window.dataLayer || [];
+                function gtag(){dataLayer.push(arguments);}
+                gtag('js', new Date());
+                gtag('config', '${process.env.NEXT_PUBLIC_GA_ID}', {
+                  page_path: window.location.pathname
+                });
+              `,
+              }}
+            />
+          </>
+        )}
+      </head>
```

Comment:
Let's use next's official recommendation: https://nextjs.org/docs/messages/next-script-for-ga